### ISSUE TITLE
Accessibility: page should only have one Header

### ISSUE THREAD

**Issue by shaunanoordin** [zooniverse/front-end-monorepo#3710]
## Accessibility Enhancement

Packages: `app-project`, `lib-react-components`, etc
Follows: #3704 
Verified on: 5f1e76391ea8f387c5707da16c931954e71de061

Guideline: for better accessibility, a web page should only have one `<header>`, which should be the child of the `<body>`. 

UPDATE: may require further discussion. See **Thought 3** below.

### Dev Notes

This issue was discovered during the [review](https://github.com/zooniverse/front-end-monorepo/pull/3704#pullrequestreview-1122284905) for 3704

> (1) Why has this header got a 'presentation' role?
> (2) Aha, [Using Navigation Landmarks](https://accessibility.blog.gov.uk/2016/05/27/using-navigation-landmarks/): a page should only have one `header`, which is a direct child of `body`. I assume this gives the page more than one `header`, in which case maybe remove the header here, or restructure the page so that it only has one header, as per the advice in Léonie Watson's post.

As of 5f1e763, we actually have **six** `<header>`s on the Project home page alone:

_Image: notable elements on the Planet Hunters TESS project home page. `headers` in red with role value (if any), `nav` in green, `section` in cyan._
![fem-issue-multiple-headers](https://user-images.githubusercontent.com/13952701/192790221-949ca407-3048-46b2-adee-09fcd419a7c0.png)

Of the six, only one (the top navigation bar + sign in menu, aka ZooHeader) explicitly sets [`role='presentation'`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role), meaning, yup, we have like five semantic `<header>`s running around the page.

(Side note: I also checked if there are any weird role overrides on the page, like somebody using `<div role='heading'>` instead of `<h1>`, but couldn't find any notable ones, so take that image above as a "good enough" baseline understanding for the current DOM structure)

**Thought 1:** I'm OK with ensuring there's only _one_ `<header>` per page, but the questions are (a) which component should be the `<header>`? (I assume it's the ProjectHeader) and (b) what should the other components be changed to?

(For the "Message From The Researcher" compo for example, it just makes sense to remove the `<header>` altogether since it's just wrapping a h2 for no reason. But for the ZooHeader... what, put a `div` with a [**banner**](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role) role? EDIT: Wait, I may have talked myself into this.)

**Thought 2:** er, making our main `<header>` the direct child of the `<body>` may be more of a challenge though, since it's buried under so many layers of divs, including some NextJS ones. Not sure how we'd approach this. 🤔 

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/13952701/192793028-0ebe636b-51d9-4396-af11-039f36ab16a0.png">

**Thought 3:** I decided to double check [WHATWG (under HTML "sectioning" elements)](https://html.spec.whatwg.org/multipage/sections.html#the-header-element) and [W3C (under ARIA)](https://w3c.github.io/html-aria/#el-header) to confirm what's the current best practice for headers, since the Gov.UK accessibility blog was written in 2016, with newer comments from 2017

Here's what I've gleaned, to the best of my understanding:

- You **can** have multiple `<headers>` on a page, and according to the specs, this shouldn't (❓) lead to any accessibility issues
  - However, additional headers tend to be the child of `article`s or `section`s
  - Example illustrated in the [nav element's example code.](https://html.spec.whatwg.org/multipage/sections.html#the-nav-element)
- A `<header>` has different _implicit roles_ depending on where it's positioned:
  - If it's **NOT** a (direct or non-direct?) child of `article`, `aside`, `main`, `nav` or `section`, (or an equivalent like `<div role='article'>`), then it has **role=banner,** which implies a navigational **landmark** (with _site-oriented_ content?)
  - If it **IS** a child of `article`/`section`/etc, then it has **role=generic,** which implies it has no semantic meaning.

(Correction: in an earlier edit, I mixed up when the header gets role=banner and role=generic)

What I'm getting for this is that (a) multiple `header`s may _not_ be an accessibility issue with the current specs? (my confidence level in this statement is only like 67% though) but (b) our goddang sub-sections that _do_ use those extra `header`s _should_ probably be wrapped in a `section` or something, and maybe have their h2/h1 headings reviewed, because things like "Message From Researcher" probably should be [complementary landmarks](https://www.w3.org/TR/wai-aria-1.2/#complementary).

**However...**

### Next Steps

...my understanding of the HTML specs is nothing compared to the actual experience of using some screenreader software and how _that_ treats `headers`, so I think I need to look into that more. My limited experiments with MacOS's VoiceOver has been... uh... I need to figure out how that works, let's say that.

**Comment by shaunanoordin** [comment]
Correction: in an earlier edit, I mixed up when the header gets role=banner and role=generic

So if a `header` _isn't_ a child of `section` or `main`, it gets `role=banner`. Which means that on our FEM home page, the random `header` we have for "Message From Researcher" is being treated as a _banner landmark,_ indicating _site-oriented content_ ???

...crap, OK, then I'm doubling down on what I said about wrapping those sub-components in `section`s

**Comment by eatyourgreens** [comment]
The `<header>` tags should be removed completely from those content boxes, since they already have headings anyway.

A page `<header>` is easy enough to implement, but needs some small changes to the project page templates.

Evey page should have a single `<main>` landmark too, wrapping the main content. Anything outside that can be an `<aside>`.
